### ISSUE TITLE
Build with libblosc for zarr encoding

### ISSUE THREAD

**Issue by TomAugspurger** [conda-forge/gdal-feedstock#567]
Follow up to https://github.com/conda-forge/gdal-feedstock/pull/557.

GDAL 3.4 includes a new driver for Zarr files: https://gdal.org/drivers/raster/zarr.html. The default compressor for Zarr-python is blosc. This means that by default, datasets written with xarray / Zarr can't be read by GDAL

```python
In [1]: import zarr

In [3]: root = zarr.group("test.zarr/")

In [4]: z = root.zeros("data", shape=(10, 10), chunks=(5, 5))
```

running gdalinfo (reads the metadata fine, but note that there's an error about the decompressor):

```
❯ gdalinfo test.zarr
ERROR 1: Decompressor blosc not handled
Driver: Zarr/Zarr
Files: test.zarr
Size is 512, 512
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0,  512.0)
Upper Right (  512.0,    0.0)
Lower Right (  512.0,  512.0)
Center      (  256.0,  256.0)
```

If we write another dataset with no compression, things are fine:

```python
In [5]: root = zarr.group("test-uncompressed.zarr/")

In [6]: z = root.zeros("data", shape=(10, 10), chunks=(5, 5), compressor=None)
```

which gives

```
❯ gdalinfo test-uncompressed.zarr
Driver: Zarr/Zarr
Files: test-uncompressed.zarr
Size is 10, 10
Corner Coordinates:
Upper Left  (    0.0,    0.0)
Lower Left  (    0.0,   10.0)
Upper Right (   10.0,    0.0)
Lower Right (   10.0,   10.0)
Center      (    5.0,    5.0)
Band 1 Block=5x5 Type=Float64, ColorInterp=Undefined
  NoData Value=0
```

I haven't built GDAL from source, but I think we need to pass a flag saying to include libblosc when building.

**Comment by xylar** [comment]
We'll see how this goes but I'm trying to add `libblosc` in #568.  Suggestions welcome.

**Comment by xylar** [comment]
@TomAugspurger, could you turn your test above into something we could add to https://github.com/conda-forge/gdal-feedstock/blob/master/recipe/run_test.py in #568?

**Comment by TomAugspurger** [comment]
Yep, I'll add something like that (possibly avoiding python-zarr if I can).

**Comment by xylar** [comment]
There's no problem with using python zarr.  We would just have to add it as a dependency during testing.

**Comment by TomAugspurger** [comment]
@xylar I made a PR to your PR at https://github.com/xylar/gdal-feedstock/pull/1.

We should maybe not merge that PR until Zarr is available for Python 3.10, which I think is blocked by https://github.com/conda-forge/numcodecs-feedstock/pull/78.

**Comment by xylar** [comment]
Ah, thanks.  You're right that tests presumably won't pass until we have zarr for python 3.10.  That's fine.  We can wait a bit.
